src/test/kotlin/reminder/application/ReminderSearcherTest.kt

```diff
@@ -0,0 +1,54 @@
+package reminder.application
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.mockito.Mockito
+import reminder.domain.Reminder
+import reminder.domain.ReminderIdentifier
+import reminder.domain.ReminderIdentifierGenerator
+import reminder.domain.ReminderRepository
+import reminder.mothers.IdentifierMother
+import reminder.mothers.ReminderMother
+import java.util.*
+import kotlin.test.assertEquals
+
+class ReminderSearcherTest {
+    private lateinit var reminderSaver: ReminderSaver
+    private lateinit var generator: ReminderIdentifierGenerator
+    private lateinit var reminderSearcher: ReminderSearcher
+    private lateinit var repository: ReminderRepository
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(ReminderRepository::class.java)
+        generator = Mockito.mock(ReminderIdentifierGenerator::class.java)
+        reminderSearcher = ReminderSearcher(repository)
+        reminderSaver = ReminderSaver(repository, generator)
+    }
+
+    @Test
+    fun `Nothing is returned when valid identifier doesn't exist in the repository`() {
+        val identifier = IdentifierMother.getValidIdentifier()
+        `Assert that the reminder was not found in the repository given the identifier`(identifier)
```

Comment:
If we only use one auxiliar test function once, why we would bother to create it? I think it is more readable have a testing method with:
```kotlin
fun dummyTest() {
 // Initialize
 // Execute
 // Assert
}
```

Instead of:
```kotlin
fun dummyTest() {
 // Initialize part of the test
  functionB()
}
fun functionB() {
 // More inits...
 // Execute
 // Assert
}
```